src/core/thread/network_data_local.cpp

```diff
@@ -111,29 +151,74 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     }
 #endif
 
-    error =
-        AddPrefix(aConfig.GetPrefix(), NetworkDataTlv::kTypeBorderRouter, aConfig.mPreference, flags, aConfig.mStable);
-
-exit:
-    return error;
+    return AddPrefix(aConfig.GetPrefix(), NetworkDataTlv::kTypeBorderRouter, aConfig.mPreference, flags,
+                     aConfig.mStable);
 }
 
 Error Local::RemoveOnMeshPrefix(const Ip6::Prefix &aPrefix)
 {
     return RemovePrefix(aPrefix, NetworkDataTlv::kTypeBorderRouter);
 }
 
+Error Local::ValidateHasRoutePrefix(const ExternalRouteConfig &aConfig)
+{
+    Error error = kErrorNone;
+
+    // Validate Prefix
+    VerifyOrExit((aConfig.GetPrefix().GetLength() > 0) && aConfig.GetPrefix().IsValid(), error = kErrorInvalidArgs);
+
+    // Validate Preference
+    switch (aConfig.mPreference)
+    {
+    case OT_ROUTE_PREFERENCE_LOW:
+    case OT_ROUTE_PREFERENCE_MED:
+    case OT_ROUTE_PREFERENCE_HIGH:
+        break;
+    default:
+        ExitNow(error = kErrorInvalidArgs);
+    }
+
+    // Check if address already configured
+    VerifyOrExit(!aConfig.GetPrefix().ContainsPrefix(Get<Mle::MleRouter>().GetMeshLocalPrefix()),
+                 error = kErrorAlready);
```

Comment:
This part of check is common for both `ValidateOnMeshPrefix` and  `ValidateHasRoutePrefix`. (used to be in the `AddPrefix` method). 

Should use a same method (e.g. `ValidatePrefix`) for these logic to reduce code size?